core/src/main/scalajvm/sttp/client3/HttpClientBackend.scala

```diff
@@ -75,9 +77,10 @@ abstract class HttpClientBackend[F[_], S, P, B](
       resBody: Either[B, WebSocket[F]],
       request: Request[T, R]
   ): F[Response[T]] = {
-    val headersMap = res.headers().map().asScala
+    val headersMap = filterEmptyContentEncoding(res.headers().map().asScala)
```

Comment:
I think I'd leave the headers as they are received from the client, but instead move the filtering to our invocations of `customEncodingHandler`. That way the user gets the headers that were sent by the server, and we don't try to decode using an empty encoding

Reply:
Yeah, the required filtering got much simplified by that, thanks a lot for pointing that out